express-api/src/services/properties/propertiesServices.ts

```diff
@@ -571,6 +572,30 @@ export type BulkUploadRowResult = {
   reason?: string;
 };
 
+const checkForHeaders = (sheetObj: Record<string, any>[]) => {
+  const requiredHeaders = [
+    'PropertyType',
+    'PID',
+    'Classification',
+    'AgencyCode',
+    'AdministrativeArea',
+    'Latitude',
+    'Longitude',
+  ];
+  for (let rowNum = 0; rowNum < sheetObj.length; rowNum++) {
+    const row = sheetObj[rowNum];
+    if (row.PropertyType == 'Building') {
+      requiredHeaders.push('Name', 'PredominateUse', 'ConstructionType');
+      break;
+    }
+  }
+  for (let rowNum = 0; rowNum < requiredHeaders.length; rowNum++) {
+    if (!Object.keys(sheetObj[0]).includes(requiredHeaders[rowNum])) {
+      throw new ErrorWithCode('Please ensure all required headers are present', 400);
+    }
```

Comment:
Should we display which headers are missing if some are not present in the sheet?

Reply:
I added a specific message depending on which header is missing. Think that's good or do we show all at once?